### PR TITLE
[ci.yaml] Set targets names equal to builder

### PR DESCRIPTION
**Pull request by CaseyHillers**
This is to deduplicate the builder field as it's no longer necessary.

Fixed some issues where there were duplicate targets for both presubmit and postsubmit.

https://github.com/flutter/flutter/issues/84998

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -9,31 +9,31 @@ enabled_branches:
   - master
 
 targets:
-  - name: linux_analyze
+  - name: Linux analyze
```
Comment (christopherfujino) [inline .ci.yaml:12]: so are target names unused now?

Comment (CaseyHillers) [reply .ci.yaml:12]: Yes (I forgot to cc you in https://github.com/flutter/cocoon/pull/1265). For posterity, the plan is to temporarily use builder name, once this has soaked, switch back to name, then deprecate the builder field.

Comment (christopherfujino) [reply .ci.yaml:12]: cool, SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**